src/app/beer_garden/requests.py

```diff
@@ -753,12 +706,31 @@ def cancel_request(request: Request) -> Request:
         ModelValidationError: The Request is already completed
 
     """
-    if request.status in Request.COMPLETED_STATUSES:
-        raise ModelValidationError("Cannot cancel a completed request")
+    request = db.query_unique(Request, id=request_id)
 
     request.status = "CANCELED"
     request = db.update(request)
 
     # TODO - Metrics here?
 
     return request
+
+
+def handle_event(event):
+    # Only care about downstream garden
+    if event.garden != beer_garden.config.get("garden.name"):
+
+        if event.name == Events.REQUEST_CREATED.name:
+            if db.query_unique(Request, id=event.payload.id) is None:
+                db.create(event.payload)
+
+        elif event.name in (Events.REQUEST_STARTED.name, Events.REQUEST_COMPLETED.name):
+            # When we send child requests to child gardens where the parent was on
+            # the local garden we remove the parent before sending them. Only setting
+            # the subset of fields that change "corrects" the parent
+            existing_request = db.query_unique(Request, id=event.payload.id)
+
+            for field in ("status", "output", "error_class"):
+                setattr(existing_request, field, getattr(event.payload, field))
```

Comment:
This addresses my one comment above, now that I understand how you are handling the events. This should maintain the parent field, without the child needing to know it was a spawned request.